### ISSUE TITLE
False positive unsubscriptable-object and no-member for subclasses of List[]

### ISSUE THREAD

**Issue by skinnersBoxy** [PyCQA/pylint#3200]
### Steps to reproduce
```python
from typing import List

class A:
    pass

class B(List[A]):
    def f(self, i):
        return self[i]  # pylint: unsubscriptable-object

b = B()
b.append(A())  # pylint: no-member
b.f(0)
```

### Current behavior
False positive warnings where indicated

### Expected behavior
No warnings

### pylint --version output
pylint 2.4.2
astroid 2.3.1
Python 3.8.0 (default, Oct 16 2019, 19:42:50) 
[GCC 8.3.0]

**Comment by PCManticore** [comment]
I'm going to close this as a potential duplicate of https://github.com/PyCQA/pylint/issues/3129. We simply do not understand `typing` constructs well enough to figure out that they are not unsubscriptable.
